crates/core/c8y_api/src/http_proxy.rs

```diff
@@ -102,6 +124,24 @@ impl C8yEndPoint {
         }
         false
     }
+
+    pub fn get_device_type(&self, device_id: Option<String>) -> DeviceType {
+        match device_id {
+            Some(device_id) => DeviceType::ChildDevice(device_id),
+            None => DeviceType::MainDevice,
+        }
+    }
```

Comment:
I would not do that because the APIs have to be changed. I wanted to keep the `http_proxy` APIs as it is and do the mapping here. From the API pov, it's calling either for the `main device` or for the `child device`. 

Other way would be we must change the APIs to get the `device-id` always, irrespective of its for a child or the main device. For example for the `send_software_list_http()`` API, this must always take a `device-id`. That way things will be clearer.

Reply:
At some point we need to change those APIs to complete the cleanup what we've started here. The problem is that some of those APIs still accept request structs like `UploadConfigFile`, `UploadLogBinary` etc that has an `Option` field for the `device-id` which can be `None` for the main device. We should use the new `DeviceType` enum to those structs as well, so that we use it across the entire call stack and avoid this `None` magic completely.

Reply:
@PradeepKiruvale I encourage you to follow the direction given by @albinsuresh